wetterdienst/provider/noaa/ghcn/api.py

```diff
@@ -0,0 +1,325 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2018-2021, earthobservations developers.
+# Distributed under the MIT License. See LICENSE for more info.
+import datetime as dt
+from enum import Enum
+from io import BytesIO
+from typing import List, Optional, Union
+
+import numpy as np
+import pandas as pd
+import pytz
+import requests
+from timezonefinder import TimezoneFinder
+
+from wetterdienst.core.scalar.request import ScalarRequestCore
+from wetterdienst.core.scalar.values import ScalarValuesCore
+from wetterdienst.metadata.columns import Columns
+from wetterdienst.metadata.datarange import DataRange
+from wetterdienst.metadata.kind import Kind
+from wetterdienst.metadata.period import Period, PeriodType
+from wetterdienst.metadata.provider import Provider
+from wetterdienst.metadata.resolution import Resolution, ResolutionType
+from wetterdienst.metadata.timezone import Timezone
+from wetterdienst.provider.noaa.ghcn.parameter import (
+    PARAMETER_MULTIPLICATION_FACTORS,
+    NoaaGhcnParameter,
+)
+from wetterdienst.provider.noaa.ghcn.unit import NoaaGhcnUnit
+from wetterdienst.util.cache import CacheExpiry, metaindex_cache
+from wetterdienst.util.network import download_file
+
+
+class NoaaGhcnDatasetBase(Enum):
```

Comment:
My own linter configuration requires a docstring fro all functions and classes. I would suggest to add them here as well. At least a short description is fine.